Makefile.am

```diff
@@ -0,0 +1,77 @@
+ACLOCAL_AMFLAGS="-Im4"
+
+SUBDIRS = chat contrib pppd pppstats pppdump
+
+if WITH_PLUGINS
+SUBDIRS += pppd/plugins
+endif
+
+DIST_SUBDIRS = $(SUBDIRS) common include modules scripts
+
+#
+# *HACK*
+#   This is to work around the kernel module for PPP on Sun Solaris
+if SUNOS
+all-am:
+	(cd solaris ; $(MAKE) -f Makefile)
+
+clean-generic:
+	(cd solaris ; $(MAKE) -f Makefile clean)
+
+install-am:
+	(cd solaris ; $(MAKE) -f Makefile install)
+endif
```

Comment:
btw, any reason not to just do
```
if SUNOS
SUBDIRS += solaris
endif
```
?

Reply:
I've tried this, and it somehow require a Automake file in that directory.

Reply:
oh sure, but can't you replace the `Makefile` in there too?

Reply:
The Makefile in that directory is non-automake based and given that it builds a kernel module for SunOS, it's incompatible with automake..

You are welcome to sync out my fork and try to change it yourself.

Reply:
This is really just for the SunOS folks that is interested in building the kernel module (and the previous build system already had that wired in), It could be decoupled from the autoconf stuff and documented in a read me file somewhere on how to build that separately. The ideal thing is to just support it as it used to and compile it inline with the automake stuff.

Reply:
Now, ideally .. if you sync out my fork and give this a spin. I hope the initial hurdle of setting up the configure process and setting the right paths help shake out a few bugs. You'd be also able to make additional suggestions where you see necessary.